File: 1026. Maximum Difference Between Node and Ancestor.java
class Solution {
int maxdiff;
public int maxAncestorDiff(TreeNode root) {
if(root == null)return 0;
    maxdiff = Integer.MIN_VALUE;
    dfs(root, root.val, root.val);
    return maxdiff;
}
private void dfs(TreeNode root, int min , int max)
{
    if(root == null)return;
    maxdiff = Math.max( maxdiff,  Math.max(Math.abs( root.val - min ), Math.abs(max - root.val)));    
    dfs(root.left,Math.min(min,root.val),Math.max(max,root.val));//check for left part    
    dfs(root.right,Math.min(min,root.val),Math.max(max,root.val));//check for right part
                       
}
}

File: 1026. Maximum Difference Between Node and Ancestor.cpp
class Solution {
public:
    int maxDiff = INT_MIN ;
    int maxAncestorDiff(TreeNode* root) {      
        if(root == NULL) return 0 ;
        findMax(root , root ) ;
        maxAncestorDiff(root->left) ;
        maxAncestorDiff(root->right) ;
        return maxDiff ;
    }
    void findMax(TreeNode* ancestor , TreeNode* child ) {
        if(ancestor == NULL) return ;
        if(child == NULL) return ;
        int difference = abs(child->val -  ancestor->val ) ;
        maxDiff = max(maxDiff , difference) ;
        findMax(ancestor , child->left) ;
        findMax(ancestor , child->right) ;
    }
};
